auto-farm/src/user_farm_actions.rs

```diff
@@ -0,0 +1,71 @@
+use common_structs::PaymentsVec;
+use farm::base_functions::{ClaimRewardsResultType, ClaimRewardsResultWrapper};
+
+use crate::farm_external_storage_read::State;
+
+elrond_wasm::imports!();
+
+#[elrond_wasm::module]
+pub trait UserFarmActionsModule:
+    crate::common_storage::CommonStorageModule
+    + crate::farms_whitelist::FarmsWhitelistModule
+    + crate::farm_external_storage_read::FarmExternalStorageReadModule
+    + crate::user_farm_tokens::UserFarmTokensModule
+    + utils::UtilsModule
+{
+    /// Arg: user to claim rewards for
+    #[endpoint(claimAllFarmRewards)]
+    fn claim_all_farm_rewards(&self, user: ManagedAddress) {
+        self.require_caller_proxy_claim_address();
+
+        let farms_mapper = self.farm_ids();
+        let user_id = self.user_ids().get_id_or_insert(&user);
+        let user_tokens_mapper = self.user_farm_tokens(user_id);
+
+        let user_tokens = user_tokens_mapper.get();
+        let mut new_user_farm_tokens = PaymentsVec::new();
```

Comment:
What is `new_user_farm_tokens` used for?

Reply:
After every farm action, user receives a new farm token, so we need to save an updated list with the new tokens. Additionally, farm tokens that have no associated farm address (which can happen if a farm is removed or paused) will be left unchanged.